vortex-ffi/src/sink.rs

```diff
@@ -0,0 +1,93 @@
+use std::ffi::CStr;
+use std::ptr::null_mut;
+
+use mpsc::Sender;
+use tokio::fs::File;
+use tokio::sync::mpsc;
+use tokio::task::JoinHandle;
+use tokio_stream::wrappers::ReceiverStream;
+use vortex::ArrayRef;
+use vortex::dtype::DType;
+use vortex::error::{VortexExpect, VortexResult, vortex_err};
+use vortex::file::VortexWriteOptions;
+use vortex::stream::ArrayStreamAdapter;
+
+use crate::RUNTIME;
+use crate::array::vx_array;
+use crate::error::{try_or, vx_error};
+use crate::file::vx_file_create_options;
+
+#[allow(non_camel_case_types)]
+/// The `sink` interface is used to collect array chunks and place them into a resource
+/// (e.g. an array stream or file (`vx_file_array_sink_create`)).
+pub struct vx_array_sink {
```

Comment:
This works for now, in future you may want to hold a `Box<dyn ArraySink>` so the logic isn't tied to file writing

Reply:
I was thinking we could introduce an enum later